jetstream/analysis.py

```diff
@@ -157,209 +158,283 @@ def _calculate_metrics(
 
         limits = TimeLimits.for_single_analysis_window(
             last_date_full_data=end_date.strftime("%Y-%m-%d"),
             analysis_start_days=0,
             analysis_length_dates=analysis_length_dates,
-            first_enrollment_date=self.config.experiment.start_date.strftime("%Y-%m-%d"),
+            first_enrollment_date=config.experiment.start_date.strftime("%Y-%m-%d"),
             num_dates_enrollment=dates_enrollment,
         )
 
         exp = mozanalysis.experiment.Experiment(
-            experiment_slug=self.config.experiment.normandy_slug,
-            start_date=self.config.experiment.start_date.strftime("%Y-%m-%d"),
+            experiment_slug=config.experiment.normandy_slug,
+            start_date=config.experiment.start_date.strftime("%Y-%m-%d"),
         )
 
         metrics = set()
-        for v in self.config.metrics.values():
+        for v in config.metrics.values():
             metrics |= {m.metric for m in v}
 
         sql = exp.build_query(
             metrics,
             limits,
             "normandy",
-            self.config.experiment.enrollment_query,
+            config.experiment.enrollment_query,
         )
 
         dry_run_query(sql)
 
-    def run(self, current_date: datetime, dry_run: bool = False) -> None:
+    def _save_statistics(
+        self,
+        period: AnalysisPeriod,
+        segment_results: List[Dict[str, Any]],
+        metrics_table: str,
+        normandy_slug: str,
+    ):
+        """Write statistics to BigQuery."""
+        job_config = bigquery.LoadJobConfig()
+        job_config.schema = StatisticResult.bq_schema
+        job_config.write_disposition = bigquery.job.WriteDisposition.WRITE_TRUNCATE
+
+        # wait for the job to complete
+        self.bigquery.load_table_from_json(
+            segment_results, f"statistics_{metrics_table}", job_config=job_config
+        )
+
+        self._publish_view(period, normandy_slug=normandy_slug, table_prefix="statistics")
+
+    def run(
+        self, current_date: datetime, config: AnalysisConfiguration, dry_run: bool = False
+    ) -> None:
         """
         Run analysis using mozanalysis for a specific experiment.
         """
-        logger.info("Analysis.run invoked for experiment %s", self.config.experiment.normandy_slug)
+        logger.info("Analysis.run invoked for experiment %s", config.experiment.normandy_slug)
+
+        self.check_runnable(config, current_date)
 
-        self.check_runnable(current_date)
+        # dask config
+        results = []
+        calculate_metrics = dask.delayed(self._calculate_metrics)
+        calculate_statistics = dask.delayed(self._calculate_statistics)
+        segment_metrics = dask.delayed(self._segment_metrics)
+        missing_counts = dask.delayed(self._missing_counts)
+        counts = dask.delayed(self._counts)
+        table_to_dataframe = dask.delayed(self.bigquery.table_to_dataframe)
+        save_statistics = dask.delayed(self._save_statistics)
```

Comment:
`dask.delayed` cannot be nested, so I ended up rewriting `_calculate_statistics` to unnest the loops calculating statistics for each segment